src/features/usersSlice.ts

```diff
@@ -0,0 +1,54 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+type UsersState = {
+  users: User[],
+  loaded: boolean,
+  hasError: string,
```

Comment:
The name `hasError` for a string variable can be misleading. It seems like it should be a boolean variable. Consider renaming it to `errorMessage` or `error`.